File: cpp/1816.truncate-sentence.cpp
/*
 * @lc app=leetcode id=1816 lang=cpp
 *
 * [1816] Truncate Sentence
 */
#include <string>
using namespace std;
// @lc code=start
class Solution
{
public:
    string truncateSentence(string s, int k)
    {
        string res;
        int word_cnt = 0;
        for (auto c : s)
        {
            if (isblank(c))
            {
                ++word_cnt;
                if (word_cnt == k)
                {
                    break;
                }
            }
            res.push_back(c);
        }
        return res;
    }
};
// @lc code=end

File: 1816.truncate-sentence.py
#
# @lc app=leetcode id=1816 lang=python3
#
# [1816] Truncate Sentence
#

# @lc code=start
class Solution:
    def truncateSentence(self, s: str, k: int) -> str:
        ind = 0
        while k:
            k = k-1
            ind = s.find(' ', ind)
            if ind != -1:
                if k:
                    ind += 1
            else:
                ind = len(s)
                break
        return s[0:ind]
# @lc code=end
